precompiles/ArbGasInfo.go

```diff
@@ -159,9 +159,38 @@ func (con ArbGasInfo) GetPricesInArbGas(c ctx, evm mech) (huge, huge, huge, erro
 // GetGasAccountingParams gets the rollup's speed limit, pool size, and block gas limit
 func (con ArbGasInfo) GetGasAccountingParams(c ctx, evm mech) (huge, huge, huge, error) {
 	l2pricing := c.State.L2PricingState()
-	speedLimit, _ := l2pricing.SpeedLimitPerSecond()
+
+	var (
+		speedLimit uint64
+		err        error
+	)
+
+	// For ArbOS version 50 and above, the speed limit is defined by the highest-period constraint
+	if c.State.ArbOSVersion() >= params.ArbosVersion_50 {
```

Comment:
```suggestion
	if c.State.ArbOSVersion() >= l2pricing.ArbosMultiConstraintsVersion {
```

It would be nice to have a constant to give a bit more context.

```
const ArbosMultiConstraintsVersion = params.ArbosVersion_50
```